### PR TITLE
feat(distributed): add pld.tile.remote_store subview push op

### PR DESCRIPTION
**Pull request by YunjiQin**
## Summary

- Adds `pld.tile.remote_store`: a tile-level cross-rank push op that writes a local tile into a sub-region of a peer rank's window-bound `DistributedTensor`. Mirror of `pld.tile.remote_load`, lowers via the same peer-addressing helpers (`CommRemoteOffset` + `addptr` + `make_tensor_view` + `partition_view`) with the final emit flipped from `tload` to `tstore`. Unlike `pld.tensor.put` (whole-tensor only), this lets cross-rank pushes target a sub-region of a larger remote tensor.
- Fixes a pre-existing offsets-lowering gap in `remote_load` / `notify` / `wait` / `remote_store` codegen: `GetExprCodes` returned raw SSA names, so runtime i32 scalars in offsets reached `pto.partition_view` (which expects index) and PTOAS rejected the kernel. Routed all four call sites through `LowerTupleToIndexSSA`.
- Refactors the L3 EP dispatch+combine ST to match the runtime example 1:1 (per-rank signal cells, dynamic `pl.range` loops, host-backed staged outputs, golden replay of dispatch).

## Changes

**New op — `pld.tile.remote_store`:**
- IR op + verifier in `src/ir/op/distributed/remote_store.cpp`
- Codegen `MakeRemoteStoreCodegenPTO` in `src/backend/common/pto_ops_common.cpp`
- Python IR builder in `python/pypto/ir/op/distributed/tile_ops.py`
- DSL wrapper in `python/pypto/language/distributed/op/tile_ops.py`
- Re-exports in the three `pld` `__init__` / `unified_ops` files

**Test refactor — `test_l3_ep_dispatch_combine.py`:**
- Switched every `pl.unroll` to `pl.range` (runtime uses dynamic loops).
- Resized barrier signals to `[N_RANKS, 1]` with per-src signal cells (`count_done_sig[N]` / `data_done_sig[N]` / `combine_done_sig[N]`); dropped the `peer` scalar param.
- Added stage_out phase: dispatch emits host-backed `recv_x_out` / `recv_w_out` / `recv_idx_out` / `recv_count_out`; local_expert and combine read from these staged outputs instead of windows.
- Replaced `pl.array.create(1, ...) / acc[0]` scalar accumulator boilerplate with plain `pl.const(0, ...)` + reassignment (SSA pass auto-promotes to loop-carried IterArg).
- Added `_compute_golden_recv` host replay to verify per-rank `recv_x` / `recv_w` / `recv_idx` / `recv_count` in addition to the combined `routed_y` check.

## Testing

- [x] `tests/st/distributed/test_l3_remote_store.py` — ring-shuffle (whole-tile push) + subview-halves (two disjoint subview writes per rank)
- [x] `tests/st/distributed/test_l3_ep_dispatch_combine.py` — 1:1 port of `runtime/examples/workers/l3/ep_dispatch_combine`; skipped on `*sim` platforms pending a PTOAS tail-emission fix (see skip reason)

### CODE REVIEW

File: src/backend/common/pto_ops_common.cpp [hunk 1]
```diff
@@ -2311,6 +2310,63 @@ static std::string MakeRemoteLoadCodegenPTO(const CallPtr& op, codegen::CodegenB
...
+  if (auto w = As<ir::ConstInt>(valid_shape[1])) width_dim = std::to_string(w->value_);
+
+  auto peer_view = EmitCommRemoteView(binding, op->args_[2], codegen);
+
+  const std::string dtype_str = codegen.GetTypeString(binding.type->dtype_);
+  std::string partition_type = MakePartitionTensorViewType({height_dim, width_dim}, dtype_str);
+  std::string partition_view = EmitPartitionViewPTO(
+      binding.var->name_hint_ + "_peer", peer_view.ssa, peer_view.view_type_str, partition_type,
+      LowerTupleToIndexSSA(offsets_tuple, codegen), {height_code, width_code}, codegen);
+
```
[the remark this answers is not in the document]
Comment (YunjiQin) [reply src/backend/common/pto_ops_common.cpp:2383]: Fixed in `cffac44b` — mirrored `tile.store`'s optional 4th shapes tuple. The op now accepts `pld.tile.remote_store(src_tile, target, peer, offsets[, shapes])` (4 or 5 args). When `shapes` is present, codegen uses it for `partition_view` sizes (matches `tile.store`'s `FlattenTileNdTo2D` injection path); when absent, it falls back to the existing 2D `tile.valid_shape` path. `tile.store` itself does not constrain target rank in its verifier either — both ops now have the same N-D codegen contract.

File: tests/st/distributed/test_l3_ep_dispatch_combine.py [hunk 2]
```diff
@@ -0,0 +1,693 @@
...
+``[src, 0]`` for every ``src != my_rank``. Matches ``count_done_sig[N]`` /
+``data_done_sig[N]`` / ``combine_done_sig[N]`` in the C++ kernel and
+generalizes to ``N_RANKS > 2``.
+
+**Stage-out — 1:1 with runtime.** The dispatch kernel emits four host-backed
+outputs: ``recv_x_out [L*R, D] BF16``, ``recv_w_out [L, R] FP32``,
+``recv_idx_out [L, R] INT32``, and ``recv_count_out [L, 1] INT32``. Per-row
+1×D tile copies for x; scalar GM reads of column 0 for w / idx (the wide
+window was filled as ``[value, 0, …, 0]`` so column 0 is the real payload).
+Downstream kernels read from the staged outputs, not the window.
```
[the remark this answers is not in the document]
Comment (YunjiQin) [reply tests/st/distributed/test_l3_ep_dispatch_combine.py:55]: Fixed in `cffac44b` — replaced both `×` occurrences (lines 53 and 247) with ASCII `x`.

File: tests/st/distributed/test_l3_remote_store.py [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+        expected = torch.stack([inputs[1], inputs[0]])
+        assert torch.allclose(outputs, expected), (
+            f"ring remote_store mismatch: max diff = {(outputs - expected).abs().max().item()}"
+        )
+
+    @pytest.mark.skip(
+        reason="Flaky: PTOAS does not drain MTE pipes before pto.comm.tnotify, "
+        "so the second remote tstore can be in flight when the signal lands on peer. "
+        "Tracked upstream at hw-native-sys/PTOAS#711."
+    )
```
[the remark this answers is not in the document]
Comment (YunjiQin) [reply tests/st/distributed/test_l3_remote_store.py]: Fixed in `cffac44b` — converted `pytest.mark.skip` to `pytest.mark.xfail(strict=False, reason=...)` so the subview `remote_store` codegen path still runs in CI. Original reason (PTOAS#711) is preserved in the xfail `reason`.

**Review by lyfne123 — APPROVED** [review]
